src/hotspot/share/services/attachListener.cpp

```diff
@@ -95,7 +210,7 @@ static InstanceKlass* load_and_initialize_klass(Symbol* sh, TRAPS) {
   return ik;
 }
 
-static jint get_properties(AttachOperation* op, outputStream* out, Symbol* serializePropertiesMethod) {
+static void get_properties(AttachOperation* op, attachStream* out, Symbol* serializePropertiesMethod) {
```

Comment:
In this function you have to remember to set the result, otherwise you'll accidentally drop the result. If you return `jint`, you're forced to remember to write a  `return`.

I'd say that you should revert the changes for this function, and instead have the callers of this method set the result code.

In fact, revert this kind of change for all functions in `AttachOperationFunctionInfo funcs[]` and have all send the return code. That simplifies the pattern significantly, as you only need to set the result early if you want to be streaming your output. That also minimizes the changeset, win-win.

Reply:
Updated as suggested. But github considers diff for the file as "large" :)